cluster/daemon.go

```diff
@@ -0,0 +1,72 @@
+package cluster
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/groupcache/groupcache-go/v3"
+	"github.com/groupcache/groupcache-go/v3/transport"
+	"github.com/groupcache/groupcache-go/v3/transport/peer"
+)
+
+// Daemon is an instance of groupcache
+type Daemon struct {
+	GroupCache *groupcache.Instance
```

Comment:
I will suggest we make this unexported and add to deamon all the methods we need from GroupCache. I believe the goal is to the `Daemon` to spawn an instance of GroupCache. I may be wrong